### DISCUSSION TITLE
error: 'SynchronizedBufferRTOS' does not name a type

### DISCUSSION THREAD

**Discussion by es-tronics** [pschatzmann/arduino-audio-tools#862] [Q&A] [answered]
Hi Phil,
First of all thank you for all these fantastic audio tools!
I'm using a ESP32-WROOM-32E to send directions to my helmet headset.	
After updating the libraries I get errors with AudioA2DP.h that I didn't have before.
When I try the streams-generator-a2dp.ino example I get the same errors:

D:\arduino-1.8.19\portable\sketchbook\libraries\arduino-audio-tools\src/AudioLibs/AudioA2DP.h:23:1: error: 'SynchronizedBufferRTOS' does not name a type
 SynchronizedBufferRTOS<uint8_t>a2dp_buffer{A2DP_BUFFER_SIZE * A2DP_BUFFER_COUNT, A2DP_BUFFER_SIZE, portMAX_DELAY, portMAX_DELAY};
 ^
And then 8 more error messages about: 'a2dp_buffer' was not declared in this scope

In file included from D:\arduino-1.8.19\portable\sketchbook\libraries\arduino-audio-tools\examples\examples-stream\streams-generator-a2dp\streams-generator-a2dp.ino:12:0:
D:\arduino-1.8.19\portable\sketchbook\libraries\arduino-audio-tools\src/AudioLibs/AudioA2DP.h: In member function 'bool audio_tools::A2DPStream::begin(audio_tools::A2DPConfig)':
D:\arduino-1.8.19\portable\sketchbook\libraries\arduino-audio-tools\src/AudioLibs/AudioA2DP.h:125:13: error: 'a2dp_buffer' was not declared in this scope
             a2dp_buffer.resize(cfg.bufferSize);

Does the example work for you? 
Can you tell me where this is going wrong?
Regards, Emile

**Comment by pschatzmann** [comment, marked as answer]
Did you already consider the first point of this [Wiki](https://github.com/pschatzmann/arduino-audio-tools/wiki/It's-not-working).

**Reply by es-tronics** [reply]
Hi Phil,
Thank you for your quick response.
I used an old json link in the Board Managers URL's that didn't go higher then 1.0.6.
So yes, I'm a few years behind and I'm sorry for wasting your time.
Thank you for pointing me in the right direction!
